### PR TITLE
Fix tp snipe complex

### PR DESCRIPTION
**Pull request by oversword**
Fixes https://github.com/BlockySurvival/issue-tracker/issues/83
Fixes https://github.com/BlockySurvival/issue-tracker/issues/320
Choose between this and https://github.com/minetest-mods/teleport-request/pull/51

This PR fixes the teleport conflicts by allowing for multiple teleport requests, and allowing the user to choose between them via a form

The form can also be manually brought up using the new command `/tpf` for "teleport form"

The form lists all requests made by the player and to the player, and provides buttons for accepting or denying requests, and cancelling their own requests

To make this possible, I have completely re-thought the way teleport requests are stored, the lists `tp.tpr_list`, `tp.tphr_list`, `tp.tpc_list` and `tp.tpn_list` should no longer be needed. Instead there is `request_list`, `sender_list` and `receiver_list`. Each request gets its own ID and is a unique object, the sender and receiver lists index these IDs by the senders and receivers for easier retrieval later.

This allows for multiple TP requests to exist at the same time without overwriting each other

This needs a lot of testing, effectively every function needs to be tested, or at least: `/tpr`, `/tphr`, `/tpy`, `/tpn`, `/tpc`

### CODE REVIEW

**Comment by Panquesito7** [comment]
Alright, it's been a very long time, going to test this out.

File: functions.lua [hunk 1]
```diff
@@ -56,6 +56,278 @@ local function send_message(player, message)
...
+	else
+		local y = 1
+		local request_list_formspec = ""
+		if sent_requests.count ~= 0 then
+			request_list_formspec = request_list_formspec..("label[0.2,%f;%s:]"):format(y, S("Sent by you"))
+			y = y+0.7
+			for request_id, _ in pairs(sent_requests) do
+				if request_id ~= "count" then
+					local request = request_list[request_id]
+					if request.direction == "receiver" then
```
Comment (Panquesito7) [inline functions.lua:219]: I believe area requests should be shown here as well.
Without the area requests, the formspec is kind of bugged out.

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Tested and everything else is working amazingly well (on MT 5.7.0).
Thanks so much! 🚀 It appears the only thing is the area requests on the formspec, and it should be done. 🙂

File: functions.lua (hunk 1 above)
Comment (oversword) [reply functions.lua:219]: Will try to take care of this soon 👍

Comment (Panquesito7) [reply functions.lua:219]: Cool, thanks. Let me know if you need any help with it. 🙂

**Review by Panquesito7 — APPROVED** [review]
Tested and works.
This is awesome. Thanks! 🚀

**Pull request merged**